lib/modem_info/modem_info_json.c

```diff
@@ -141,6 +146,10 @@ int modem_info_json_string_get(char *buf)
 		}
 	}
 
+	if (IS_ENABLED(CONFIG_MODEM_INFO_ADD_BOARD)) {
+		ret += json_add_str(data_obj, "BOARD", CONFIG_BOARD);
```

Comment:
It can be considered to handle the errors when they occur both here and in the loop above, instead of continuing regardless.